onnxscript/rewriter/_rewrite_rule.py

```diff
@@ -136,7 +136,59 @@ def match(
                 if var.name is not None:
                     if var.name not in match.bindings:
                         match.bind(var.name, None)
+
+            # Perform value/node level checks before condition function
             try:
+                # Check node-level checkers
+                for pattern_node, ir_node in match.node_bindings.items():
+                    if pattern_node.check_method is not None:
+                        check_result = pattern_node.check_method(context, ir_node)
+                        if not check_result:
+                            if isinstance(check_result, _basics.MatchResult):
```

Comment:
This code-pattern of what to do when the check fails is repeated thrice (here, and for value-level checking, and finally in the pre-existing down below for _condition_function. Can you extract this out as a local utility function, say "fail", so that we can simply say `return fail(...)` in all 3 places?